src/main/java/org/checkerframework/checker/androidresource/AndroidResourceAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,276 @@
+            AnnotationBuilder.fromClass(elements, FractionContainer.class);
+    protected final AnnotationMirror ID_CONTAINER =
+            AnnotationBuilder.fromClass(elements, IdContainer.class);
+    protected final AnnotationMirror INTEGER_CONTAINER =
+            AnnotationBuilder.fromClass(elements, IntegerContainer.class);
+    protected final AnnotationMirror INTERPOLATOR_CONTAINER =
+            AnnotationBuilder.fromClass(elements, InterpolatorContainer.class);
+    protected final AnnotationMirror LAYOUT_CONTAINER =
+            AnnotationBuilder.fromClass(elements, LayoutContainer.class);
+    protected final AnnotationMirror MENU_CONTAINER =
+            AnnotationBuilder.fromClass(elements, MenuContainer.class);
+    protected final AnnotationMirror PLURALS_CONTAINER =
+            AnnotationBuilder.fromClass(elements, PluralsContainer.class);
+    protected final AnnotationMirror RAW_CONTAINER =
+            AnnotationBuilder.fromClass(elements, RawContainer.class);
+    protected final AnnotationMirror STRING_CONTAINER =
+            AnnotationBuilder.fromClass(elements, StringContainer.class);
+    protected final AnnotationMirror STYLEABLE_CONTAINER =
+            AnnotationBuilder.fromClass(elements, StyleableContainer.class);
+    protected final AnnotationMirror XML_CONTAINER =
+            AnnotationBuilder.fromClass(elements, XmlContainer.class);
+
+    public AndroidResourceAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+    }
+
+    @Override
+    protected TreeAnnotator createTreeAnnotator() {
+        return new ListTreeAnnotator(
+                new AndroidResourceTreeAnnotator(this), super.createTreeAnnotator());
+    }
+
+    private class AndroidResourceTreeAnnotator extends TreeAnnotator {
+
+        public AndroidResourceTreeAnnotator(AnnotatedTypeFactory atypeFactory) {
+            super(atypeFactory);
+        }
+
+        /*
+         * Method for a tree node for a member access expression. For example:
+         *       expression . identifier
+         */
+        @Override
+        public Void visitMemberSelect(MemberSelectTree node, AnnotatedTypeMirror annotatedTypeMirror) {
+
+            String expression = node.getExpression().toString();
+            String identifier = node.getIdentifier().toString();
+
+            assignContainerAnnotations(annotatedTypeMirror, expression, identifier);
+
+            assignResAnnotations(annotatedTypeMirror, expression);
+
+//            System.out.println("Expression: " + node.getExpression() + ", Identifier: " + node.getIdentifier()
+//                    + ", ATM: " + annotatedTypeMirror.toString() + "\n");
+
+            return null;
+        }
+
+        /**
+         * Assign @XXXRes annotations to member select.
+         *
+         * @param annotatedTypeMirror
+         * @param expression
+         */
+        private void assignResAnnotations(AnnotatedTypeMirror annotatedTypeMirror, String expression) {

```

Comment:
This code is not using the `@XXXContainer` annotations to make a decision.
Again, there was a reason why we had the container annotations and you should write tests for those and then actually implement logic that uses the container annotations.

Reply:
Sorry! I will update it soon. Thanks.